src/webgpu/shader/validation/const_assert/const_assert.spec.ts

```diff
@@ -7,44 +7,174 @@ import { ShaderValidationTest } from '../shader_validation_test.js';
   .params(u =>
     u
       .combine('case', keysOf(kConditionCases))
       .combine('scope', ['module', 'function'] as const)
       .beginSubcases()
   )
   .fn(t => {
-    const constants = `
-const one = 1;
-const two = 2;
-const three = 2;
-`;
-    const expr = kConditionCases[t.params.case];
-    t.expectCompileResult(true, constants + buildStaticAssert(true, expr, t.params.scope));
-    t.expectCompileResult(false, constants + buildStaticAssert(false, expr, t.params.scope));
+    const expr = kConditionCases[t.params.case].expr;
+    const val = kConditionCases[t.params.case].val;
+    t.expectCompileResult(
+      true,
+      kConditionConstants + buildStaticAssert(val ? expr : `!(${expr})`, t.params.scope)
+    );
+  });
+
+g.test('constant_expression_assert')
+  .desc(`Test that const_assert does assert on a false conditional expression`)
+  .params(u =>
+    u
+      .combine('case', keysOf(kConditionCases))
+      .combine('scope', ['module', 'function'] as const)
+      .beginSubcases()
+  )
+  .fn(t => {
+    const expr = kConditionCases[t.params.case].expr;
+    const val = kConditionCases[t.params.case].val;
+    t.expectCompileResult(
+      false,
+      kConditionConstants + buildStaticAssert(val ? `!(${expr})` : expr, t.params.scope)
+    );
+  });
+
+g.test('constant_expression_logical_or_no_assert')
+  .desc(
+    `Test that const_assert does not assert on a condition expression that contains a logical or and evaluates true`
+  )
+  .params(u =>
+    u
+      .combine('lhs', keysOf(kConditionCases))
+      .combine('rhs', keysOf(kConditionCases))
+      .combine('scope', ['module', 'function'] as const)
+      .beginSubcases()
+  )
+  .fn(t => {
+    const expr = `(${kConditionCases[t.params.lhs].expr}) || (${
+      kConditionCases[t.params.rhs].expr
+    })`;
+    const val = kConditionCases[t.params.lhs].val || kConditionCases[t.params.rhs].val;
+    t.expectCompileResult(
+      true,
+      kConditionConstants + buildStaticAssert(val ? expr : `!(${expr})`, t.params.scope)
+    );
+  });
+
+g.test('constant_expression_logical_or_assert')
+  .desc(
+    `Test that const_assert does assert on a condition expression that contains a logical or and evaluates false`
+  )
+  .params(u =>
+    u
+      .combine('lhs', keysOf(kConditionCases))
+      .combine('rhs', keysOf(kConditionCases))
+      .combine('scope', ['module', 'function'] as const)
+      .beginSubcases()
+  )
+  .fn(t => {
+    const expr = `(${kConditionCases[t.params.lhs].expr}) || (${
+      kConditionCases[t.params.rhs].expr
+    })`;
+    const val = kConditionCases[t.params.lhs].val || kConditionCases[t.params.rhs].val;
+    t.expectCompileResult(
+      false,
+      kConditionConstants + buildStaticAssert(val ? `!(${expr})` : expr, t.params.scope)
+    );
+  });
+
+g.test('constant_expression_logical_and_no_assert')
+  .desc(
+    `Test that const_assert does not assert on a condition expression that contains a logical and and evaluates true`
```

Comment:
nit: this sentence is hard to parse. Recommend:

`Test that const_assert does not assert on a condition expression that contains a logical-and which evaluates to true`

Similarly for the other comments.

Reply:
Done